### DISCUSSION TITLE
Dumping Attributes with H5Easy to H5 Groups / Root

### DISCUSSION THREAD

**Discussion by mgaimann** [BlueBrain/HighFive#757] [Questions] [answered]
Hello,

I would like to add attributes to the root of a H5 file (which is a group and not a dataset).

When trying to do this via
```
H5Easy::dumpAttribute(h5File, "/", "seed", 12345);
```
the error
```
    if (file.getObjectType(path) != ObjectType::Dataset) {
        throw error(file, path, "H5Easy::dumpAttribute: path not a DataSet");
    }
```
appears (defined in the file H5Easy_misc.hpp). Dumping to a dataset works perfectly fine.

Is dumping attributes to groups not supported with H5Easy? Is it supported at all? (I could not find such case in the referenced [example](https://github.com/BlueBrain/HighFive/blob/master/src/examples/create_attribute_string_integer.cpp)).

What would you suggest as a solution? Creating an "empty" or "dummy" dataset at the root and dumping the attributes there?

Thank you for your help :)

**Comment by 1uc** [comment, marked as answer]
Thank you for reporting this bug. I've improved the examples to show how it can be done using core HighFive in #758. Note that a `H5Easy::File` is just a `HighFive::File`.

**Reply by mgaimann** [reply]
Thank you for the quick response!
